### DISCUSSION TITLE
Could username in the  Authentication block of the KafaConnect operator be retrievable by a secret?

### DISCUSSION THREAD

**Discussion by jvieten** [strimzi/strimzi-kafka-operator#8145] [General]
Hello,
I am using the KakaConnect Operator to authenticate against confluent.cloud.
confluet.cloud client authentication works with API keys which have random usernames (id)
in my Kubernetes cluster the confluent API keys are stored in  Secrets. Therefore It would be nice to be able to retrieve the Authentication.username too from a secret when installing the KafkaConenct operator.

```kind: KafkaConnect
metadata:
  name: kafka-connect-jdbc
  annotations:
    strimzi.io/use-connector-resources: "true"
spec:
  version: 3.3.1
 ....
  authentication:
    type: plain
    username:  {{ .Values.kafka.username }}
    passwordSecret:
      secretName: kafka-connect-secret
      password: kafka-password
```

My workaround currently is the fetch the username before running helm by kubectl.

Is this a valid to request to ask if it would be possible to make the username in the future in the  Authentication retrievable by a secret? Or are there better ways to achieve my goal ?

**Comment by scholzj** [comment]
That is not supported currently. If you would be interested to contribute it, one thing we will need to figure out would be the API and how would it look like. So a proposal would be probably needed for this. It would also need to be done in all the other places for consistency.
